### DISCUSSION TITLE
How to add new language support with antlr4 (groovy)

### DISCUSSION THREAD

**Discussion by dolphin323** [GumTreeDiff/gumtree#317] [Q&A]
Hello!

I want to add support for groovy language using antlr4. I'm using the grammar from [this parser](https://github.com/daniellansun/groovy-parser) that is the same one as the one on the [official groovy website](https://github.com/apache/groovy/tree/master/src/antlr). 

I tried mocking solutions in other forked repositories for languages such as xml (https://github.com/svanteschubert/gumtree). However, I'm facing various issues.

Can you please provide clear instructions on how to setup an ANTLR4 grammar with the GumTree tool?

**Comment by jrfaller** [comment]
Hi!

Sorry to not be helpful but I have little experience with the AntLR bridge :(. I mostly use tree-sitter to integrate new languages.
